src/components/base/Select/index.js

```diff
@@ -49,14 +49,43 @@ const Row = styled.div`
 `
 const rowHeight = 40 // Fixme We should pass this as a prop for dynamic rows?
 class MenuList extends PureComponent<*> {
+  state = {
+    children: null,
+  }
+
+  static getDerivedStateFromProps({ children }, state) {
+    if (children !== state.children) {
+      const currentIndex = Math.max(children.findIndex(({ props: { isFocused } }) => isFocused), 0)
+
+      return {
+        children,
+        currentIndex,
+      }
+    }
+    return null
+  }
+
+  componentDidMount() {
+    this.componentDidUpdate()
```

Comment:
hmm, not a big fan here.

Can you create a function that does this, and use it in `componentDidMount` and `componentDidUpdate` instead ?

Reply:
Done, I copies that and didn't even check it 😅